go/ql/lib/semmle/go/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -311,6 +311,13 @@ class DataFlowCallable extends TDataFlowCallable {
         .hasLocationInfo(filepath, startline, startcolumn, endline, endcolumn)
   }
 
+  /** Gets the location of this callable. */
+  Location getLocation() {
+    result = getCallableLocation(this.asCallable()) or
+    result = this.asFileScope().getLocation() or
```

Comment:
As an aside, can I complain that the dataflow library assumes that every method/function has an enclosing callable, so we have to add file scope as a kind of callable? Could that be made less confusing for languages with file-scoped functions, perhaps with a different scope type that could be set as the same as the callable type in java?

Reply:
Good question. I think this mostly comes down to naming, which is inherently hard. We could call it something like "scope", but then it's less clear that that actually means method/function most of the time, and `CallableOrFileScope` is a bit of a mouthful - especially when the data flow library doesn't need to know about the distinction between callables and file-scopes.